ranges.cpp

```diff
@@ -484,6 +507,214 @@ constexpr auto transform(Func p) {
 }
 /******************************************/
 
+/******************************************/
+/* startAt_view                         */
+/* Iterates over all elements of a view   */
+/* starting in the first element which   */
+/* matches the function, and ending in	  */
+/* the element just before the matching one    */
+/******************************************/
+template<typename View>
+class startAt_view : public view_base {
+public:
+    class iterator;
+    using startAt_func_t = std::function<bool(typename View::value_type&)>;
+    using value_type = typename View::value_type;
+    using pointer = value_type*;
+    using reference = value_type&;
+
+private:
+    View m_view{};
+    startAt_func_t m_func;
+    iterator m_begin, m_end;
+    std::optional<iterator> m_start, m_finish;
+
+public:
+    // Circular iterator
+    class iterator {
+    public: // <-- important
+        using difference_type = typename View::iterator::difference_type;
+        using value_type = typename View::value_type;
+        using pointer = value_type*;
+        using reference = value_type&;
+        using iterator_category	= std::forward_iterator_tag;
+
+        startAt_view* m_view{};
+    private:
+        typename View::iterator m_iter{};
+
+        std::optional<bool> m_bInitialized;
+
+    public:
+        //constexpr iterator() = default;
+        constexpr iterator(typename View::iterator iter, startAt_view* f) :  m_iter{ iter }, m_view{ f } {
+        }
+        
+        constexpr void SetInit(bool bInit){
```

Comment:
```suggestion
        constexpr void setInit(bool bInit){
```
For a unified naming style